<!DOCTYPE html>
<html lang="en" dir="ltr" class="client-nojs">
<head>
<meta charset="UTF-8" />
<title>GFF3 - GMOD</title>
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />
<meta name="generator" content="MediaWiki 1.23.13" />
<link rel="alternate" type="application/rss+xml" title="GMOD News - RSS 2.0" href="/mediawiki/index.php?title=Special:NewsChannel&amp;format=rss20" />
<link rel="alternate" type="application/atom+xml" title="GMOD News - Atom 1.0" href="/mediawiki/index.php?title=Special:NewsChannel&amp;format=atom10" />
<link rel="ExportRDF" type="application/rdf+xml" title="GFF3" href="/mediawiki/index.php?title=Special:ExportRDF/GFF3&amp;xmlmime=rdf" />
<link rel="shortcut icon" href="/favicon.ico" />
<link rel="search" type="application/opensearchdescription+xml" href="/mediawiki/opensearch_desc.php" title="GMOD (en)" />
<link rel="EditURI" type="application/rsd+xml" href="http://gmod.org/mediawiki/api.php?action=rsd" />
<link rel="copyright" href="http://www.gnu.org/licenses/fdl-1.3.html" />
<link rel="alternate" type="application/atom+xml" title="GMOD Atom feed" href="/mediawiki/index.php?title=Special:RecentChanges&amp;feed=atom" />
<link rel="stylesheet" href="http://gmod.org/mediawiki/load.php?debug=false&amp;lang=en&amp;modules=mediawiki.legacy.commonPrint%2Cshared%7Cmediawiki.skinning.interface%7Cmediawiki.ui.button%7Cskins.vector.styles&amp;only=styles&amp;skin=vector&amp;*" />
<meta name="ResourceLoaderDynamicStyles" content="" />
<link rel="stylesheet" href="http://gmod.org/mediawiki/load.php?debug=false&amp;lang=en&amp;modules=site&amp;only=styles&amp;skin=vector&amp;*" />
<style>a:lang(ar),a:lang(kk-arab),a:lang(mzn),a:lang(ps),a:lang(ur){text-decoration:none}
/* cache key: wikidb:resourceloader:filter:minify-css:7:b36fd1c042133c9c9b60260f7c29b237 */</style>
<script src="http://gmod.org/mediawiki/load.php?debug=false&amp;lang=en&amp;modules=startup&amp;only=scripts&amp;skin=vector&amp;*"></script>
<script>if(window.mw){
mw.config.set({"wgCanonicalNamespace":"","wgCanonicalSpecialPageName":false,"wgNamespaceNumber":0,"wgPageName":"GFF3","wgTitle":"GFF3","wgCurRevisionId":27236,"wgRevisionId":27236,"wgArticleId":3940,"wgIsArticle":true,"wgIsRedirect":false,"wgAction":"view","wgUserName":null,"wgUserGroups":["*"],"wgCategories":["GFF3","Tutorials","2013 Summer School"],"wgBreakFrames":false,"wgPageContentLanguage":"en","wgPageContentModel":"wikitext","wgSeparatorTransformTable":["",""],"wgDigitTransformTable":["",""],"wgDefaultDateFormat":"dmy","wgMonthNames":["","January","February","March","April","May","June","July","August","September","October","November","December"],"wgMonthNamesShort":["","Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"],"wgRelevantPageName":"GFF3","wgIsProbablyEditable":false,"wgRestrictionEdit":[],"wgRestrictionMove":[],"sfgAutocompleteValues":[],"sfgAutocompleteOnAllChars":false,"sfgFieldProperties":[],"sfgDependentFields":[],"sfgShowOnSelect":[],"sfgScriptPath":"/mediawiki/extensions/SemanticForms","sdgDownArrowImage":"/mediawiki/extensions/SemanticDrilldown/skins/down-arrow.png","sdgRightArrowImage":"/mediawiki/extensions/SemanticDrilldown/skins/right-arrow.png"});
}</script><script>if(window.mw){
mw.loader.implement("user.options",function($,jQuery){mw.user.options.set({"ccmeonemails":0,"cols":80,"date":"default","diffonly":0,"disablemail":0,"editfont":"default","editondblclick":0,"editsectiononrightclick":0,"enotifminoredits":0,"enotifrevealaddr":0,"enotifusertalkpages":1,"enotifwatchlistpages":1,"extendwatchlist":0,"fancysig":0,"forceeditsummary":0,"gender":"unknown","hideminor":0,"hidepatrolled":0,"imagesize":2,"math":1,"minordefault":0,"newpageshidepatrolled":0,"nickname":"","norollbackdiff":0,"numberheadings":0,"previewonfirst":0,"previewontop":1,"rcdays":7,"rclimit":50,"rows":25,"showhiddencats":0,"shownumberswatching":1,"showtoolbar":1,"skin":"vector","stubthreshold":0,"thumbsize":2,"underline":2,"uselivepreview":0,"usenewrc":0,"watchcreations":1,"watchdefault":1,"watchdeletion":0,"watchlistdays":3,"watchlisthideanons":0,"watchlisthidebots":0,"watchlisthideliu":0,"watchlisthideminor":0,"watchlisthideown":0,"watchlisthidepatrolled":0,"watchmoves":0,"wllimit":250,
"useeditwarning":1,"prefershttps":1,"language":"en","variant-gan":"gan","variant-iu":"iu","variant-kk":"kk","variant-ku":"ku","variant-shi":"shi","variant-sr":"sr","variant-tg":"tg","variant-uz":"uz","variant-zh":"zh","searchNs0":true,"searchNs1":false,"searchNs2":false,"searchNs3":false,"searchNs4":false,"searchNs5":false,"searchNs6":false,"searchNs7":false,"searchNs8":false,"searchNs9":false,"searchNs10":false,"searchNs11":false,"searchNs12":false,"searchNs13":false,"searchNs14":false,"searchNs15":false,"searchNs102":false,"searchNs103":false,"searchNs104":false,"searchNs105":false,"searchNs106":false,"searchNs107":false,"searchNs108":false,"searchNs109":false,"searchNs170":false,"searchNs171":false,"searchNs274":false,"searchNs275":false,"variant":"en"});},{},{});mw.loader.implement("user.tokens",function($,jQuery){mw.user.tokens.set({"editToken":"+\\","patrolToken":false,"watchToken":false});},{},{});
/* cache key: wikidb:resourceloader:filter:minify-js:7:c46df80add0ce29ec25bae6c3f9c06b1 */
}</script>
<script>if(window.mw){
mw.loader.load(["ext.smw.style","ext.smw.tooltips","mediawiki.page.startup","mediawiki.legacy.wikibits","mediawiki.legacy.ajax","skins.vector.js"]);
}</script>
<meta name="google-site-verification" content="2Wkz2pqjhCA65_bPal0zYi0KpOeVsAchA2XdVvfI8bA" /><script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(["_setAccount", "UA-33095045-1"]);
  _gaq.push(["_trackPageview"]);

  (function() {
    var ga = document.createElement("script"); ga.type = "text/javascript"; ga.async = true;
    ga.src = ("https:" == document.location.protocol ? "https://ssl" : "http://www") + ".google-analytics.com/ga.js";
    var s = document.getElementsByTagName("script")[0]; s.parentNode.insertBefore(ga, s);
  })();

</script><!--[if lt IE 7]><style type="text/css">body{behavior:url("/mediawiki/skins/vector/csshover.min.htc")}</style><![endif]--></head>
<body class="mediawiki ltr sitedir-ltr ns-0 ns-subject page-GFF3 skin-vector action-view vector-animateLayout">
		<div id="mw-page-base" class="noprint"></div>
		<div id="mw-head-base" class="noprint"></div>
		<div id="content" class="mw-body" role="main">
			<a id="top"></a>
			<div id="mw-js-message" style="display:none;"></div>
						<h1 id="firstHeading" class="firstHeading" lang="en"><span dir="auto">GFF3</span></h1>
						<div id="bodyContent">
								<div id="siteSub">From GMOD</div>
								<div id="contentSub"></div>
												<div id="jump-to-nav" class="mw-jump">
					Jump to:					<a href="#mw-navigation">navigation</a>, 					<a href="#p-search">search</a>
				</div>
				<div id="mw-content-text" lang="en" dir="ltr" class="mw-content-ltr"><table class="tutorialheader">
<tr>
<td> <span style="font-size: 160%; line-height: 120%"> <a href="/mediawiki/index.php?title=GMOD:GFF3&amp;action=edit&amp;redlink=1" class="new" title="GMOD:GFF3 (page does not exist)">GFF3</a> <a href="/mediawiki/index.php?title=Category:Session&amp;action=edit&amp;redlink=1" class="new" title="Category:Session (page does not exist)">Session</a></span><br /><br />
<p><a href="/wiki/2013_GMOD_Summer_School" title="2013 GMOD Summer School">2013 GMOD Summer School</a><br />
19-23 July 2013<br />
<a href="/wiki/User:Scott" title="User:Scott">Scott Cain</a>
</p>
</td>
<td align="right"> <a href="/wiki/File:2013-summer-school.png" class="image"><img alt="2013-summer-school.png" src="/mediawiki/images/thumb/f/fc/2013-summer-school.png/200px-2013-summer-school.png" width="200" height="124" srcset="/mediawiki/images/thumb/f/fc/2013-summer-school.png/300px-2013-summer-school.png 1.5x, /mediawiki/images/thumb/f/fc/2013-summer-school.png/400px-2013-summer-school.png 2x" /></a>
</td></tr></table>
<p><br />
<b>GFF</b> is a standard file format for storing genomic features in a text file.  GFF stands for <i>Generic Feature Format</i>.  GFF files are plain text, 9 column, tab-delimited files.  GFF <i><a href="/wiki/Databases_and_GMOD" title="Databases and GMOD">databases</a></i> also exist.  They use a <a href="/wiki/Glossary#Schema" title="Glossary">schema</a> custom built to represent GFF data.  GFF is <a href="#GFF_in_GMOD">frequently used in GMOD</a> for data exchange and representation of genomic data.
</p>
<div id="toc" class="toc"><div id="toctitle"><h2>Contents</h2></div>
<ul>
<li class="toclevel-1 tocsection-1"><a href="#Versions"><span class="tocnumber">1</span> <span class="toctext">Versions</span></a></li>
<li class="toclevel-1 tocsection-2"><a href="#GFF3"><span class="tocnumber">2</span> <span class="toctext">GFF3</span></a>
<ul>
<li class="toclevel-2 tocsection-3"><a href="#GFF3_Annotation_Section"><span class="tocnumber">2.1</span> <span class="toctext">GFF3 Annotation Section</span></a>
<ul>
<li class="toclevel-3 tocsection-4"><a href="#GFF3_Format"><span class="tocnumber">2.1.1</span> <span class="toctext">GFF3 Format</span></a></li>
<li class="toclevel-3 tocsection-5"><a href="#Nesting_Features"><span class="tocnumber">2.1.2</span> <span class="toctext">Nesting Features</span></a></li>
<li class="toclevel-3 tocsection-6"><a href="#Discontinuous_Features"><span class="tocnumber">2.1.3</span> <span class="toctext">Discontinuous Features</span></a></li>
<li class="toclevel-3 tocsection-7"><a href="#Protein-Coding_Genes"><span class="tocnumber">2.1.4</span> <span class="toctext">Protein-Coding Genes</span></a></li>
<li class="toclevel-3 tocsection-8"><a href="#Alignments"><span class="tocnumber">2.1.5</span> <span class="toctext">Alignments</span></a></li>
<li class="toclevel-3 tocsection-9"><a href="#Quantitative_Data"><span class="tocnumber">2.1.6</span> <span class="toctext">Quantitative Data</span></a></li>
</ul>
</li>
<li class="toclevel-2 tocsection-10"><a href="#GFF3_Sequence_Section"><span class="tocnumber">2.2</span> <span class="toctext">GFF3 Sequence Section</span></a></li>
<li class="toclevel-2 tocsection-11"><a href="#GFF3_Validation"><span class="tocnumber">2.3</span> <span class="toctext">GFF3 Validation</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-12"><a href="#GFF2"><span class="tocnumber">3</span> <span class="toctext">GFF2</span></a></li>
<li class="toclevel-1 tocsection-13"><a href="#GTF"><span class="tocnumber">4</span> <span class="toctext">GTF</span></a></li>
<li class="toclevel-1 tocsection-14"><a href="#GFF_in_GMOD"><span class="tocnumber">5</span> <span class="toctext">GFF in GMOD</span></a>
<ul>
<li class="toclevel-2 tocsection-15"><a href="#Apollo"><span class="tocnumber">5.1</span> <span class="toctext">Apollo</span></a></li>
<li class="toclevel-2 tocsection-16"><a href="#Chado"><span class="tocnumber">5.2</span> <span class="toctext">Chado</span></a></li>
<li class="toclevel-2 tocsection-17"><a href="#CMap"><span class="tocnumber">5.3</span> <span class="toctext">CMap</span></a></li>
<li class="toclevel-2 tocsection-18"><a href="#GBrowse"><span class="tocnumber">5.4</span> <span class="toctext">GBrowse</span></a></li>
</ul>
</li>
<li class="toclevel-1 tocsection-19"><a href="#See_Also"><span class="tocnumber">6</span> <span class="toctext">See Also</span></a></li>
</ul>
</div>

<h1><span class="mw-headline" id="Versions">Versions</span></h1>
<p>GFF has several versions, the most recent of which is <i><a href="#GFF3">GFF3</a></i>. GFF3 addresses several shortcomings in its predecessor, <i><a href="#GFF2">GFF2</a></i>.  <b>GFF3 is the preferred format in GMOD</b>, but data is not always available in GFF3 format, so you may have to use GFF2.  The two versions are similar but are not compatible and scripts usually only work with one of the other format.  This page discusses GFF3 in detail.  GFF2 details are covered on a <a href="/wiki/GFF2" title="GFF2">separate page</a>.
</p><p>Unfortunately, people, documentation, and even this web site are not always clear about what version of GFF is being discussed.  This web page will always specify which version it is referring to.
</p><p>Finally, <i><a href="#GTF">GTF</a></i> is another file format that is very similar to GFF and is sometimes referred to as GFF2.5.
</p>
<h1><span class="mw-headline" id="GFF3">GFF3</span></h1>
<p>The <a rel="nofollow" class="external text" href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md">formal specification of GFF3</a> is on the <a rel="nofollow" class="external text" href="http://www.sequenceontology.org/">Sequence Ontology</a> web site.  It completely describes the format, including column definitions, metadata and directives.  It also contains lengthy sections explaining how to represent different situations in GFF3, including:
</p>
<ul>
<li> canonical genes non-coding transcripts
</li>
<li> parent (part-of) relationships
</li>
<li> alignments
</li>
<li> ontology association and database cross references
</li>
<li> single exon genes
</li>
<li> polycistronic transcripts
</li>
<li> genes containing inteins
</li>
<li> trans-spliced transcripts
</li>
<li> programmed frameshifts
</li>
<li> operons
</li>
</ul>
<p>Some of these cases are covered on this page as well.  If you want the full and definitive explanation of GFF3 then see <a rel="nofollow" class="external text" href="http://www.sequenceontology.org/gff3.shtml">the standard</a>.
</p>
<h2><span class="mw-headline" id="GFF3_Annotation_Section">GFF3 Annotation Section</span></h2>
<p>This first describes the format of the annotation section, and then provides explanations of how to represent several different types of data.
</p>
<h3><span class="mw-headline" id="GFF3_Format">GFF3 Format</span></h3>
<p>GFF3 format is a flat tab-delimited file. The first line of the file is a comment that identifies the file format and version. This is followed by a series of data lines, each one of which corresponds to an annotation.Here is a miniature GFF3 file:
</p>
<pre>
##gff-version 3
ctg123  .  exon  1300  1500  .  +  .  ID=exon00001
ctg123  .  exon  1050  1500  .  +  .  ID=exon00002
ctg123  .  exon  3000  3902  .  +  .  ID=exon00003
ctg123  .  exon  5000  5500  .  +  .  ID=exon00004
ctg123  .  exon  7000  9000  .  +  .  ID=exon00005
</pre>
<p>The <tt>##gff-version 3</tt> line is required and <i>must</i> be the first line of the file.  It introduces the annotation section of the file.
</p><p>The 9 columns of the annotation section are as follows:
</p><p><b>Column 1:</b> "seqid"
</p>
<dl>
<dd>The ID of the landmark used to establish the coordinate system for the current feature. IDs may contain any characters, but must escape any characters not in the set [a-zA-Z0-9.:^*$@!+_?-|].  In particular, IDs may not contain unescaped whitespace and must not begin with an unescaped "&gt;".
</dd>
</dl>
<dl>
<dd>To escape a character in this, or any of the other GFF3 fields, replace it with the percent sign followed by its hexadecimal representation. For example, "&gt;" becomes "%E3". See <a rel="nofollow" class="external text" href="http://www.blooberry.com/indexdot/html/topics/urlencoding.htm">URL Encoding (or: 'What are those "%20" codes in URLs?')</a> for details.
</dd>
</dl>
<p><b>Column 2:</b> "source"
</p>
<dl>
<dd>The source is a free text qualifier intended to describe the algorithm or operating procedure that generated this feature.  Typically this is the name of a piece of software, such as "Genescan" or a database name, such as "Genbank."  In effect, the source is used to extend the feature ontology by adding a qualifier to the type creating a new composite type that is a subclass of the type in the type column. It is not necessary to specify a source. If there is no source, put a "." (a period) in this field.
</dd>
</dl>
<p><b>Column 3:</b> "type"
</p>
<dl>
<dd>The type of the feature (previously called the "method").  This is constrained to be either: (a) a term from the "lite" sequence ontology, SOFA; or (b) a SOFA accession number.  The latter alternative is distinguished using the syntax SO:000000. This field is required.
</dd>
</dl>
<p><b>Columns 4 &amp; 5:</b> "start" and "end"
</p>
<dl>
<dd>The start and end of the feature, in 1-based integer coordinates, relative to the landmark given in column 1.  Start is always less than or equal to end.
</dd>
</dl>
<dl>
<dd>For zero-length features, such as insertion sites, start equals end and the implied site is to the right of the indicated base in the direction of the landmark. These fields are required.
</dd>
</dl>
<p><b>Column 6:</b> "score"
</p>
<dl>
<dd>The score of the feature, a floating point number.  As in earlier versions of the format, the semantics of the score are ill-defined. It is strongly recommended that E-values be used for sequence similarity features, and that P-values be used for ab initio gene prediction features. If there is no score, put a "." (a period) in this field.
</dd>
</dl>
<p><b>Column 7:</b> "strand"
</p>
<dl>
<dd>The strand of the feature.  + for positive strand (relative to the landmark), - for minus strand, and . for features that are not stranded.  In addition,&#160;? can be used for features whose strandedness is relevant, but unknown.
</dd>
</dl>
<p><b>Column 8:</b> "phase"
</p>
<dl>
<dd>For features of type "CDS", the phase indicates where the feature begins with reference to the reading frame.  The phase is one of the integers 0, 1, or 2, indicating the number of bases that should be removed from the beginning of this feature to reach the first base of the next codon. In other words, a phase of "0" indicates that the next codon begins at the first base of the region described by the current line, a phase of "1" indicates that the next codon begins at the second base of this region, and a phase of "2" indicates that the codon begins at the third base of this region. This is NOT to be confused with the frame, which is simply start modulo 3. If there is no phase, put a "." (a period) in this field.
</dd>
</dl>
<dl>
<dd>For forward strand features, phase is counted from the start field. For reverse strand features, phase is counted from the end field.
</dd>
</dl>
<dl>
<dd>The phase is <b>required</b> for all CDS features.
</dd>
</dl>
<p><b>Column 9:</b> "attributes"
</p>
<dl>
<dd>A list of feature attributes in the format tag=value.  Multiple tag=value pairs are separated by semicolons.  URL escaping rules are used for tags or values containing the following characters: ",=;". Spaces are allowed in this field, but tabs must be replaced with the&#160;%09 URL escape. This field is not required.
</dd>
</dl>
<div class="indent">
<p><b>Column 9 Tags</b>
</p><p>Column 9 tags have predefined meanings:
</p>
<dl>
<dt>ID
</dt>
<dd>Indicates the unique identifier of the feature. IDs must be unique within the scope of the GFF file.
</dd>
</dl>
<dl>
<dt>Name
</dt>
<dd>Display name for the feature.  This is the name to be displayed to the user.  Unlike IDs, there is no requirement that the Name be unique within the file.
</dd>
</dl>
<dl>
<dt>Alias
</dt>
<dd>A secondary name for the feature.  It is suggested that this tag be used whenever a secondary identifier for the feature is needed, such as locus names and accession numbers.  Unlike ID, there is no requirement that Alias be unique within the file.
</dd>
</dl>
<dl>
<dt>Parent
</dt>
<dd>Indicates the parent of the feature.  A parent ID can be used to group exons into transcripts, transcripts into genes, and so forth.  A feature may have multiple parents. Parent can *only* be used to indicate a partof relationship.
</dd>
</dl>
<dl>
<dt>Target
</dt>
<dd>Indicates the target of a nucleotide-to-nucleotide or protein-to-nucleotide alignment.  The format of the value is "target_id start end [strand]", where strand is optional and may be "+" or "-".  If the target_id contains spaces, they must be escaped as hex escape&#160;%20.
</dd>
</dl>
<dl>
<dt>Gap
</dt>
<dd>The alignment of the feature to the target if the two are not collinear (e.g. contain gaps).  The alignment format is taken from the CIGAR format described in the Exonerate documentation.  <a rel="nofollow" class="external free" href="http://cvsweb.sanger.ac.uk/cgi-bin/cvsweb.cgi/exonerate?cvsroot=Ensembl">http://cvsweb.sanger.ac.uk/cgi-bin/cvsweb.cgi/exonerate?cvsroot=Ensembl</a>).  See the <a rel="nofollow" class="external text" href="http://www.sequenceontology.org/gff3.shtml">GFF3 specification</a> for more information.
</dd>
</dl>
<dl>
<dt>Derives_from
</dt>
<dd>Used to disambiguate the relationship between one feature and another when the relationship is a temporal one rather than a purely structural "part of" one.  This is needed for polycistronic genes.  See the <a rel="nofollow" class="external text" href="http://www.sequenceontology.org/gff3.shtml">GFF3 specification</a> for more information.
</dd>
</dl>
<dl>
<dt>Note
</dt>
<dd>A free text note.
</dd>
</dl>
<dl>
<dt>Dbxref
</dt>
<dd>A database cross reference.  See the <a rel="nofollow" class="external text" href="http://www.sequenceontology.org/gff3.shtml">GFF3 specification</a> for more information.
</dd>
</dl>
<dl>
<dt>Ontology_term
</dt>
<dd>A cross reference to an ontology term.  See the <a rel="nofollow" class="external text" href="http://www.sequenceontology.org/gff3.shtml">GFF3 specification</a> for more information.
</dd>
</dl>
<p>Multiple attributes of the same type are indicated by separating the values with the comma "," character, as in:
</p>
<pre>Parent=AF2312,AB2812,abc-3
</pre>
<p>Note that attribute names are case sensitive.  "Parent" is not the same as "parent".
</p><p>All attributes that begin with an uppercase letter are reserved for later use.  Attributes that begin with a lowercase letter can be used freely by applications. You can stash any semi-structured data into the database  by using one or more unreserved (lowercase) tags.
</p>
</div>
<h3><span class="mw-headline" id="Nesting_Features">Nesting Features</span></h3>
<p>Many genomic features are discontinuous and have multiple subparts. GFF3 represents such features by linking the parts together with the Parent tag. For example, to represent an mRNA transcript that has five exons, we could write this:
</p>
<pre>##gff-version 3
ctg123 . mRNA            1300  9000  .  +  .  ID=mrna0001;Name=sonichedgehog
ctg123 . exon            1300  1500  .  +  .  ID=exon00001;Parent=mrna0001
ctg123 . exon            1050  1500  .  +  .  ID=exon00002;Parent=mrna0001
ctg123 . exon            3000  3902  .  +  .  ID=exon00003;Parent=mrna0001
ctg123 . exon            5000  5500  .  +  .  ID=exon00004;Parent=mrna0001
ctg123 . exon            7000  9000  .  +  .  ID=exon00005;Parent=mrna0001
</pre>
<p>The first feature is an mRNA that extends from position 1300 to 9000 in genomic coordinates. It has an ID of "mrna0001" and a human-readable name of "sonichedgehog" (note that the ID and the Name are <b>not</b> the same thing). This is followed by five exon features, each of which is linked to the mRNA using a Parent tag. When <a href="/wiki/GBrowse" title="GBrowse">GBrowse</a> displays this transcript, it will display each of the exons linked together by a solid line. The entire set can be found by searching for the name "sonichedgehog."
</p><p>The ID is really only important for linking features together. If a feature does not have any subparts, then it does not formally need an ID. Thus, we could simplify this by removing all the exon IDs:
</p>
<pre>##gff-version 3
ctg123 . mRNA            1300  9000  .  +  .  ID=mrna0001;Name=sonichedgehog
ctg123 . exon            1300  1500  .  +  .  Parent=mrna0001
ctg123 . exon            1050  1500  .  +  .  Parent=mrna0001
ctg123 . exon            3000  3902  .  +  .  Parent=mrna0001
ctg123 . exon            5000  5500  .  +  .  Parent=mrna0001
ctg123 . exon            7000  9000  .  +  .  Parent=mrna0001
</pre>
<p>Multiple levels of nesting are allowed. If this transcript is part of an operon, then we can add another level of nesting:
</p>
<pre>##gff-version 3
ctg123 . operon          1300 15000  .  +  .  ID=operon001;Name=superOperon
ctg123 . mRNA            1300  9000  .  +  .  ID=mrna0001;Parent=operon001;Name=sonichedgehog
ctg123 . exon            1300  1500  .  +  .  Parent=mrna0001
ctg123 . exon            1050  1500  .  +  .  Parent=mrna0001
ctg123 . exon            3000  3902  .  +  .  Parent=mrna0001
ctg123 . exon            5000  5500  .  +  .  Parent=mrna0001
ctg123 . exon            7000  9000  .  +  .  Parent=mrna0001
ctg123 . mRNA           10000 15000  .  +  .  ID=mrna0002;Parent=operon001;Name=subsonicsquirrel
ctg123 . exon           10000 12000  .  +  .  Parent=mrna0002
ctg123 . exon           14000 15000  .  +  .  Parent=mrna0002
</pre>
<h3><span class="mw-headline" id="Discontinuous_Features">Discontinuous Features</span></h3>
<p>In addition to nested features, another common type of genomic annotation is the <i>discontinuous feature</i> in which a single feature spans multiple discontinuous portions of the genome. The primary example is an alignment, such as a cDNA sequence that has been aligned to genomic sequence. GFF3 deals with these features by representing each continuous segment as a distinct row, and then giving each segment the same ID to tie them together. For example:
</p>
<pre>ctg123 example match 26122 26126 . + . ID=match001
ctg123 example match 26497 26869 . + . ID=match001
ctg123 example match 27201 27325 . + . ID=match001
ctg123 example match 27372 27433 . + . ID=match001
ctg123 example match 27565 27565 . + . ID=match001
</pre>
<p>Note that this is distinct from the nested features we looked at in the previous section. In the former case, there is a single parent feature and multiple child features that are linked to the parent via a Parent tag. The IDs of the children are distinct from each other (or absent altogether). In the latter case, each segment of the discontinuous feature has the same ID. There is no parent.
</p><p><i>Note that this method of grouping discontinuous features is not currently supported by the GMOD Chado bulk GFF3 loader.  Parent-child grouping is required.</i>
</p>
<h3><span class="mw-headline" id="Protein-Coding_Genes">Protein-Coding Genes</span></h3>
<p>We'll now look at how to represent several common cases, starting with protein-coding genes.
</p><p>The most general way of representing a protein-coding gene is the so-called "three-level gene." The top level is a feature of type "gene" which bundles up the gene's transcripts and regulatory elements. Beneath this level are one or more transcripts of type "mRNA". This level can also accommodate promoters and other cis-regulatory elements. At the third level are the components of the mRNA transcripts, most commonly CDS coding segments and UTRs. This example shows how to represent a gene named "EDEN" which has three alternatively-spliced mRNA transcripts:
</p>
<pre>ctg123 example gene            1050 9000 . + . ID=EDEN;Name=EDEN;Note=protein kinase

ctg123 example mRNA            1050 9000 . + . ID=EDEN.1;Parent=EDEN;Name=EDEN.1;Index=1
ctg123 example five_prime_UTR  1050 1200 . + . Parent=EDEN.1
ctg123 example CDS             1201 1500 . + 0 Parent=EDEN.1
ctg123 example CDS             3000 3902 . + 0 Parent=EDEN.1
ctg123 example CDS             5000 5500 . + 0 Parent=EDEN.1
ctg123 example CDS             7000 7608 . + 0 Parent=EDEN.1
ctg123 example three_prime_UTR 7609 9000 . + . Parent=EDEN.1

ctg123 example mRNA            1050 9000 . + . ID=EDEN.2;Parent=EDEN;Name=EDEN.2;Index=1
ctg123 example five_prime_UTR  1050 1200 . + . Parent=EDEN.2
ctg123 example CDS             1201 1500 . + 0 Parent=EDEN.2
ctg123 example CDS             5000 5500 . + 0 Parent=EDEN.2
ctg123 example CDS             7000 7608 . + 0 Parent=EDEN.2
ctg123 example three_prime_UTR 7609 9000 . + . Parent=EDEN.2

ctg123 example mRNA            1300 9000 . + . ID=EDEN.3;Parent=EDEN;Name=EDEN.3;Index=1
ctg123 example five_prime_UTR  1300 1500 . + . Parent=EDEN.3
ctg123 example five_prime_UTR  3000 3300 . + . Parent=EDEN.3
ctg123 example CDS             3301 3902 . + 0 Parent=EDEN.3
ctg123 example CDS             5000 5500 . + 1 Parent=EDEN.3
ctg123 example CDS             7000 7600 . + 1 Parent=EDEN.3
ctg123 example three_prime_UTR 7601 9000 . + . Parent=EDEN.3
</pre>
<p>We start with a feature of type "gene" with the ID "EDEN". This has three alternative splice forms named EDEN.1, EDEN.2 and EDEN.3. To tell <a href="/wiki/GBrowse" title="GBrowse">GBrowse</a> that each of these splice forms are part of the same gene, we give each one a Parent attribute of "EDEN" corresponding to the ID of the parent gene. Now consider mRNA EDEN.1. It has a five_prime_UTR feature, a three_prime_UTR feature, and four CDS features. To indicate that the CDS and UTR features belong to the mRNA, we give the mRNA a unique ID of "EDEN.1" and give each of the subfeatures a corresponding parent. This pattern repeats for each of the other two splice forms. Note how the five_prime_UTR of EDEN.3 is split in two parts.
</p><p>We use "Name" to give the gene and its alternative splice forms a human-readable name, and use Note to provide a description for the gene as a whole (you can add notes to the individual mRNAs but they won't display by default). The Index=1 attribute is a hint to some indexed database to make the mRNAs searchable by name. This lets users find the gene by searching for the mRNA names ("EDEN.1") as well as by the gene name ("EDEN"). However, it is usually unnecessary to do this. Also notice that we are using the Phase column for the CDS features to describe how the CDS is translated into protein. See the description of phase at the beginning of this section.
</p><p>There are other ways of representing genes. Please see <a rel="nofollow" class="external text" href="http://www.sequenceontology.org/gff3.shtml">the GFF3 Specification</a> and <a rel="nofollow" class="external text" href="http://gmod.svn.sourceforge.net/viewvc/gmod/Generic-Genome-Browser/branches/stable/docs/tutorial/tutorial.html?content-type=text%2Fhtml">The GBrowse Administration Tutorial</a> for more information.
</p>
<h3><span class="mw-headline" id="Alignments">Alignments</span></h3>
<p>Nucleotide to genome, and protein to genome alignments are a little tricky because they involve two coordinate systems, the coordinates of the alignment on the genome (known as the "source" coordinates), and the coordinates of the cDNA, EST or protein (known as the "target" coordinates). In GFF3, the target coordinates are specified using the <b>Target</b> tag.
</p>
<pre>ctg123 est EST_match 1050 1500 . + . ID=Match1;Name=agt830.5;Target=agt830.5 1 451
ctg123 est EST_match 3000 3202 . + . ID=Match1;Name=agt830.5;Target=agt830.5 452 654

ctg123 est EST_match 5410 5500 . - . ID=Match2;Name=agt830.3;Target=agt830.3 505 595
ctg123 est EST_match 7000 7503 . - . ID=Match2;Name=agt830.3;Target=agt830.3 1 504

ctg123 est EST_match 1050 1500 . + . ID=Match3;Name=agt221.5;Target=agt221.5 1 451
ctg123 est EST_match 5000 5500 . + . ID=Match3;Name=agt221.5;Target=agt221.5 452 952
ctg123 est EST_match 7000 7300 . + . ID=Match3;Name=agt221.5;Target=agt221.5 953 1253
</pre>
<p>This example shows three different alignment features of type "EST_match". Each alignment has a distinct ID, and all the discontinuous parts of the alignment have the same ID, as described earlier. In addition to the ID and Name tags, each segment also has a Target tag whose value has the format "&lt;target seqid&gt; &lt;target start&gt; &lt;target end&gt;." For example, the very first line indicates that the EST named agt830.5 aligns to genomic contig ctg123 such that positions 1 through 451 of agt830.5 aligns to bases 1050-1500 of ctg123.
</p><p>Using the ##FASTA section of the GFF3 file, you can specify the sequence of the ESTs as well as of the contig, and <a href="/wiki/GBrowse" title="GBrowse">GBrowse</a> will display the DNA and/or protein sequences in the appropriate contexts.
</p><p>See the <a rel="nofollow" class="external text" href="http://www.sequenceontology.org/gff3.shtml">GFF3 specification</a> for instructions on how to represent gapped alignments.
</p>
<h3><span class="mw-headline" id="Quantitative_Data">Quantitative Data</span></h3>
<p>GBrowse can plot quantitative data such as alignment scores, confidence scores from gene prediction programs, and microarray intensity data. There is a simple format that can be placed directly inside of a GFF3 file but does not scale to very large data sets, and a "WIG" format designed for very high-density quantitative data such as tiling arrays.
</p><p>We first look at the simple format:
</p>
<pre>ctg123 affy microarray_oligo   1 100 281 . . Name=Expt1
ctg123 affy microarray_oligo 101 200 183 . . Name=Expt1
ctg123 affy microarray_oligo 201 300 213 . . Name=Expt1
ctg123 affy microarray_oligo 301 400 191 . . Name=Expt1
ctg123 affy microarray_oligo 401 500 288 . . Name=Expt1
ctg123 affy microarray_oligo 501 600 184 . . Name=Expt1
</pre>
<p>In this format, which can be embedded directly in the GFF3 file, each data point is a distinct feature with a start and end point. The features are grouped together by giving them a common experimental name so that they can be retrieved together. We use the <b>score</b> field (column 6) to represent the quantitative information (e.g. hybridization intensity).
</p><p>In contrast, when using WIG format, the quantitative data is kept outside of the main database in a special-purpose binary file that is kept somewhere on the file system. In this case the GFF3 file contains a single line per experiment like this one:
</p>
<pre>ctg123 . microarray_oligo 1 50000 . . . Name=example;wigfile=/usr/data/ctg123.Expt1.wig
</pre>
<p>The .wig file is created and managed using a script called <tt>wiggle2gff3.pl</tt> that comes with <a href="/wiki/GBrowse" title="GBrowse">GBrowse</a>. Instructions on how to use this script is described in the <a rel="nofollow" class="external text" href="http://gmod.cvs.sourceforge.net/*checkout*/gmod/Generic-Genome-Browser/docs/tutorial/tutorial.html?pathrev=stable">GBrowse Administration Tutorial</a>.
</p>
<h2><span class="mw-headline" id="GFF3_Sequence_Section">GFF3 Sequence Section</span></h2>
<p><strong class="selflink">GFF3</strong> files can also include sequence in FASTA format at the end of the file.  The FASTA sequences are preceded by a <tt>##FASTA</tt> line. This sequence section is optional.  If present, the sequence section can define sequence for any landmark used in column 1 (the frame of reference).  For example:
For example:
</p>
<pre>
##gff-version 3
ctg123 . exon            1300  1500  .  +  .  ID=exon00001
ctg123 . exon            1050  1500  .  +  .  ID=exon00002
ctg123 . exon            3000  3902  .  +  .  ID=exon00003
ctg123 . exon            5000  5500  .  +  .  ID=exon00004
ctg123 . exon            7000  9000  .  +  .  ID=exon00005
##FASTA
&gt;ctg123
cttctgggcgtacccgattctcggagaacttgccgcaccattccgccttg
tgttcattgctgcctgcatgttcattgtctacctcggctacgtgtggcta
tctttcctcggtgccctcgtgcacggagtcgagaaaccaaagaacaaaaa
aagaaattaaaatatttattttgctgtggtttttgatgtgtgttttttat
aatgatttttgatgtgaccaattgtacttttcctttaaatgaaatgtaat
cttaaatgtatttccgacgaattcgaggcctgaaaagtgtgacgccattc
...
</pre>
<p>When the GFF3 file is processed the IDs on the header line of FASTA entries are matched with IDs used in column 1 in the annotation section of the file.
</p><p>You don't have to store the FASTA in the GFF file.  You can also store your sequences in a separate file containing only FASTA entries.
</p>
<h2><span class="mw-headline" id="GFF3_Validation">GFF3 Validation</span></h2>
<p>You can validate reasonably large GFF3 files at the following sites:
</p>
<ul>
<li> <a rel="nofollow" class="external text" href="http://modencode.oicr.on.ca/cgi-bin/validate_gff3_online">modENCODE validator</a>
</li>
</ul>
<p>The validator code can be found in the <a rel="nofollow" class="external text" href="http://sourceforge.net/p/gmod/svn/HEAD/tree/gff-validator/">GMOD Sourceforge SVN repository</a>.
</p>
<h1><span class="mw-headline" id="GFF2">GFF2</span></h1>
<p><a href="/wiki/GFF2" title="GFF2">GFF2</a> is a supported format in GMOD, <b>but it is now deprecated and if you have a choice you should use GFF3</b>.  Unfortunately, data is sometimes only available in GFF2 format.  GFF2 has a number of shortcomings compared to GFF3.
</p><p>See <a href="/wiki/GFF2" title="GFF2">GFF2</a> for more on this format.
</p>
<h1><span class="mw-headline" id="GTF">GTF</span></h1>
<p><a rel="nofollow" class="external text" href="http://mblab.wustl.edu/GTF2.html"><i>GTF</i></a>, is another file format that is very similar to <a href="/wiki/GFF2" title="GFF2">GFF2</a> and is sometimes referred to as GFF2.5.  GTF is not a supported format in GMOD so if you have a GTF file you'll need to convert it to <a href="#GFF3">GFF3</a>.  The <tt><a rel="nofollow" class="external text" href="http://song.cvs.sourceforge.net/song/software/scripts/gtf2gff3/">gft2gff3</a></tt> script does this conversion, with some caveats.  See also <a rel="nofollow" class="external text" href="http://www.nabble.com/Hi-td17810093.html">this BioPerl-l posting</a>.
</p>
<h1><span class="mw-headline" id="GFF_in_GMOD">GFF in GMOD</span></h1>
<p>A number of <a href="/wiki/GMOD_Components" title="GMOD Components">GMOD Components</a> support GFF files.  This section provides a brief description of that support.
</p>
<h2><span class="mw-headline" id="Apollo">Apollo</span></h2>
<p>The <a href="/wiki/Apollo" title="Apollo">Apollo</a> genome annotation editor can read and write annotations in GFF3 format.  You can also load GFF3 data into <a href="/wiki/Chado" title="Chado" class="mw-redirect">Chado</a> and have Apollo connect with the database.
</p>
<h2><span class="mw-headline" id="Chado">Chado</span></h2>
<p>GFF3 data can be loaded into and dumped from <a href="/wiki/Chado" title="Chado" class="mw-redirect">Chado</a> databases.  See:
</p>
<ul>
<li> <a href="/wiki/Load_GFF_Into_Chado" title="Load GFF Into Chado">Load GFF Into Chado</a>
</li>
<li> <a href="/wiki/Load_BLAST_Into_Chado" title="Load BLAST Into Chado">Load BLAST Into Chado</a> - by converting it to GFF3
</li>
<li> <a href="/wiki/Load_GenBank_into_Chado" title="Load GenBank into Chado">Load GenBank into Chado</a> - by converting it to GFF3
</li>
<li> <a href="/wiki/Load_RefSeq_Into_Chado" title="Load RefSeq Into Chado">Load RefSeq Into Chado</a> - by converting it to GFF3
</li>
<li> <a href="/wiki/Chado_Update_via_GFF" title="Chado Update via GFF">Chado Update via GFF</a>
</li>
<li> <a href="/wiki/GMODTools" title="GMODTools">GMODTools</a> - Generate <strong class="selflink">GFF3</strong> from a <a href="/wiki/Chado" title="Chado" class="mw-redirect">Chado</a> database.
</li>
</ul>
<h2><span class="mw-headline" id="CMap">CMap</span></h2>
<p>The <a href="/wiki/CMap" title="CMap">CMap</a> comparative mapping viewer can read data in GFF3 format.
</p>
<h2><span class="mw-headline" id="GBrowse">GBrowse</span></h2>
<p>The <a href="/wiki/GBrowse" title="GBrowse">GBrowse</a> genome viewer supports data in <a href="/wiki/GBrowse_Adaptors" title="GBrowse Adaptors">many formats</a>, but in many ways GFF3 is its native data format.  GBrowse also supports <a href="/wiki/GFF2" title="GFF2">GFF2</a> data.  See the <a href="/wiki/GBrowse" title="GBrowse">GBrowse</a> and <a href="/wiki/GBrowse_Adaptors" title="GBrowse Adaptors">GBrowse Adaptors</a> pages for details.
</p>
<h1><span class="mw-headline" id="See_Also">See Also</span></h1>
<ul>
<li> <a rel="nofollow" class="external text" href="http://www.sequenceontology.org/gff3.shtml">GFF3 Specification</a>
</li>
<li> <a rel="nofollow" class="external text" href="http://www.sanger.ac.uk/Software/formats/GFF/GFF_Spec.shtml">GFF2 Specification</a>
</li>
<li> <a href="/wiki/GFF2" title="GFF2">GFF2</a> - GFF2 in GMOD
</li>
<li> <a rel="nofollow" class="external text" href="http://www.broad.mit.edu/annotation/argo/help/gff.html">Broad Institute's Argo File Formats GFF page</a>
</li>
<li> <a rel="nofollow" class="external text" href="http://www.bioperl.org/wiki/GFF">BioPerl's GFF page</a>
</li>
</ul>

<!-- 
NewPP limit report
CPU time usage: 0.049 seconds
Real time usage: 0.051 seconds
Preprocessor visited node count: 118/1000000
Preprocessor generated node count: 214/1000000
Post‐expand include size: 7656/2097152 bytes
Template argument size: 30/2097152 bytes
Highest expansion depth: 2/40
Expensive parser function count: 0/100
-->

<!-- Saved in parser cache with key wikidb:pcache:idhash:3940-0!*!0!!en!2!*! and timestamp 20230327114553 and revision id 27236
 -->
</div>								<div class="printfooter">
				Retrieved from "<a href="http://gmod.org/mediawiki/index.php?title=GFF3&amp;oldid=27236">http://gmod.org/mediawiki/index.php?title=GFF3&amp;oldid=27236</a>"				</div>
												<div id='catlinks' class='catlinks'><div id="mw-normal-catlinks" class="mw-normal-catlinks"><a href="/wiki/Special:Categories" title="Special:Categories">Categories</a>: <ul><li><a href="/mediawiki/index.php?title=Category:GFF3&amp;action=edit&amp;redlink=1" class="new" title="Category:GFF3 (page does not exist)">GFF3</a></li><li><a href="/wiki/Category:Tutorials" title="Category:Tutorials">Tutorials</a></li><li><a href="/mediawiki/index.php?title=Category:2013_Summer_School&amp;action=edit&amp;redlink=1" class="new" title="Category:2013 Summer School (page does not exist)">2013 Summer School</a></li></ul></div></div>												<div class="visualClear"></div>
							</div>
		</div>
		<div id="mw-navigation">
			<h2>Navigation menu</h2>
			<div id="mw-head">
				<div id="p-personal" role="navigation" class="" aria-labelledby="p-personal-label">
	<h3 id="p-personal-label">Personal tools</h3>
	<ul>
<li id="pt-login"><a href="/mediawiki/index.php?title=Special:UserLogin&amp;returnto=GFF3" title="You are encouraged to log in; however, it is not mandatory [o]" accesskey="o">Log in / create account</a></li>	</ul>
</div>
				<div id="left-navigation">
					<div id="p-namespaces" role="navigation" class="vectorTabs" aria-labelledby="p-namespaces-label">
	<h3 id="p-namespaces-label">Namespaces</h3>
	<ul>
					<li  id="ca-nstab-main" class="selected"><span><a href="/wiki/GFF3"  title="View the content page [c]" accesskey="c">Page</a></span></li>
					<li  id="ca-talk" class="new"><span><a href="/mediawiki/index.php?title=Talk:GFF3&amp;action=edit&amp;redlink=1"  title="Discussion about the content page [t]" accesskey="t">Discussion</a></span></li>
			</ul>
</div>
<div id="p-variants" role="navigation" class="vectorMenu emptyPortlet" aria-labelledby="p-variants-label">
	<h3 id="mw-vector-current-variant">
		</h3>
	<h3 id="p-variants-label"><span>Variants</span><a href="#"></a></h3>
	<div class="menu">
		<ul>
					</ul>
	</div>
</div>
				</div>
				<div id="right-navigation">
					<div id="p-views" role="navigation" class="vectorTabs" aria-labelledby="p-views-label">
	<h3 id="p-views-label">Views</h3>
	<ul>
					<li id="ca-view" class="selected"><span><a href="/wiki/GFF3" >Read</a></span></li>
					<li id="ca-viewsource"><span><a href="/mediawiki/index.php?title=GFF3&amp;action=edit"  title="This page is protected.&#10;You can view its source [e]" accesskey="e">View source</a></span></li>
					<li id="ca-history" class="collapsible"><span><a href="/mediawiki/index.php?title=GFF3&amp;action=history"  title="Past revisions of this page [h]" accesskey="h">View history</a></span></li>
			</ul>
</div>
<div id="p-cactions" role="navigation" class="vectorMenu emptyPortlet" aria-labelledby="p-cactions-label">
	<h3 id="p-cactions-label"><span>Actions</span><a href="#"></a></h3>
	<div class="menu">
		<ul>
					</ul>
	</div>
</div>
<div id="p-search" role="search">
	<h3><label for="searchInput">Search</label></h3>
	<form action="/mediawiki/index.php" id="searchform">
					<div id="simpleSearch">
					<input type="search" name="search" placeholder="Search" title="Search GMOD [f]" accesskey="f" id="searchInput" /><input type="hidden" value="Special:Search" name="title" /><input type="submit" name="fulltext" value="Search" title="Search the pages for this text" id="mw-searchButton" class="searchButton mw-fallbackSearchButton" /><input type="submit" name="go" value="Go" title="Go to a page with this exact name if exists" id="searchButton" class="searchButton" />		</div>
	</form>
</div>
				</div>
			</div>
			<div id="mw-panel">
					<div id="p-logo" role="banner"><a style="background-image: url(http://gmod.org/images/GMOD-cogs.png);" href="/wiki/Main_Page"  title="Visit the main page"></a></div>
				<div class="portal" role="navigation" id='p-Navigation' aria-labelledby='p-Navigation-label'>
	<h3 id='p-Navigation-label'>Navigation</h3>
	<div class="body">
		<ul>
			<li id="n-GMOD-Home"><a href="/wiki/Main_Page">GMOD Home</a></li>
			<li id="n-Software"><a href="/wiki/GMOD_Components">Software</a></li>
			<li id="n-Categories-.2F-Tags"><a href="/wiki/Categories">Categories / Tags</a></li>
			<li id="n-View-all-pages"><a href="/wiki/Special:AllPages">View all pages</a></li>
		</ul>
	</div>
</div>
<div class="portal" role="navigation" id='p-Documentation' aria-labelledby='p-Documentation-label'>
	<h3 id='p-Documentation-label'>Documentation</h3>
	<div class="body">
		<ul>
			<li id="n-Overview"><a href="/wiki/Overview">Overview</a></li>
			<li id="n-FAQs"><a href="/wiki/Category:FAQ">FAQs</a></li>
			<li id="n-HOWTOs"><a href="/wiki/Category:HOWTO">HOWTOs</a></li>
			<li id="n-Glossary"><a href="/wiki/Glossary">Glossary</a></li>
		</ul>
	</div>
</div>
<div class="portal" role="navigation" id='p-Community' aria-labelledby='p-Community-label'>
	<h3 id='p-Community-label'>Community</h3>
	<div class="body">
		<ul>
			<li id="n-GMOD-News"><a href="/wiki/GMOD_News">GMOD News</a></li>
			<li id="n-Training-.2F-Outreach"><a href="/wiki/Training_and_Outreach">Training / Outreach</a></li>
			<li id="n-Support"><a href="/wiki/Support">Support</a></li>
			<li id="n-GMOD-Promotion"><a href="/wiki/GMOD_Promotion">GMOD Promotion</a></li>
			<li id="n-Meetings"><a href="/wiki/Meetings">Meetings</a></li>
			<li id="n-Calendar"><a href="/wiki/Calendar">Calendar</a></li>
		</ul>
	</div>
</div>
<div class="portal" role="navigation" id='p-tb' aria-labelledby='p-tb-label'>
	<h3 id='p-tb-label'>Tools</h3>
	<div class="body">
		<ul>
			<li id="t-whatlinkshere"><a href="/wiki/Special:WhatLinksHere/GFF3" title="A list of all wiki pages that link here [j]" accesskey="j">What links here</a></li>
			<li id="t-recentchangeslinked"><a href="/wiki/Special:RecentChangesLinked/GFF3" title="Recent changes in pages linked from this page [k]" accesskey="k">Related changes</a></li>
			<li id="t-specialpages"><a href="/wiki/Special:SpecialPages" title="A list of all special pages [q]" accesskey="q">Special pages</a></li>
			<li id="t-print"><a href="/mediawiki/index.php?title=GFF3&amp;printable=yes" rel="alternate" title="Printable version of this page [p]" accesskey="p">Printable version</a></li>
			<li id="t-permalink"><a href="/mediawiki/index.php?title=GFF3&amp;oldid=27236" title="Permanent link to this revision of the page">Permanent link</a></li>
			<li id="t-info"><a href="/mediawiki/index.php?title=GFF3&amp;action=info">Page information</a></li>
			<li id="t-smwbrowselink"><a href="/wiki/Special:Browse/GFF3" rel="smw-browse">Browse properties</a></li>
<li id="t-pdf"><a href="/mediawiki/index.php?title=Special:PdfPrint&amp;page=GFF3">Print as PDF</a></li>		</ul>
	</div>
</div>
			</div>
		</div>
		<div id="footer" role="contentinfo">
							<ul id="footer-info">
											<li id="footer-info-lastmod"> Last updated at 20:25 on 12 July 2016.</li>
											<li id="footer-info-viewcount">213,714 page views.</li>
											<li id="footer-info-copyright">Content is available under <a class="external" rel="nofollow" href="http://www.gnu.org/licenses/fdl-1.3.html">a GNU Free Documentation License</a> unless otherwise noted.</li>
									</ul>
							<ul id="footer-places">
											<li id="footer-places-about"><a href="/wiki/GMOD:About" title="GMOD:About">About GMOD</a></li>
									</ul>
										<ul id="footer-icons" class="noprint">
					<li id="footer-copyrightico">
						<a href="http://www.gnu.org/licenses/fdl-1.3.html"><img src="http://www.gnu.org/graphics/gfdl-logo-small.png" alt="a GNU Free Documentation License" width="88" height="31" /></a>
					</li>
					<li id="footer-poweredbyico">
						<a href="//www.mediawiki.org/"><img src="/mediawiki/skins/common/images/poweredby_mediawiki_88x31.png" alt="Powered by MediaWiki" width="88" height="31" /></a>
						<a href="https://www.semantic-mediawiki.org/wiki/Semantic_MediaWiki"><img src="/mediawiki/extensions/SemanticMediaWiki/includes/../resources/images/smw_button.png" alt="Powered by Semantic MediaWiki" width="88" height="31" /></a>
					</li>
				</ul>
						<div style="clear:both"></div>
		</div>
		<script>/*<![CDATA[*/window.jQuery && jQuery.ready();/*]]>*/</script><script>if(window.mw){
mw.loader.state({"site":"loading","user":"ready","user.groups":"ready"});
}</script>
<script>if(window.mw){
mw.loader.load(["mediawiki.action.view.postEdit","mediawiki.user","mediawiki.hidpi","mediawiki.page.ready","mediawiki.searchSuggest","ext.TableEdit","skins.vector.collapsibleNav"],null,true);
}</script>
<script src="http://gmod.org/mediawiki/load.php?debug=false&amp;lang=en&amp;modules=site&amp;only=scripts&amp;skin=vector&amp;*"></script>
<script>if(window.mw){
mw.config.set({"wgBackendResponseTime":112});
}</script>
	</body>
</html>
